ext/net/ops_win_pipe.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::path::Path;
+use std::rc::Rc;
+
+use deno_core::OpState;
+use deno_core::ResourceId;
+use deno_core::op2;
+use deno_permissions::OpenAccessKind;
+use deno_permissions::PermissionsContainer;
+use serde::Deserialize;
+use tokio::net::windows::named_pipe;
+
+use crate::ops::NetError;
+use crate::win_pipe::NamedPipe;
+
+fn is_true() -> bool {
+  true
+}
+
+#[derive(Deserialize, Debug)]
+pub enum PipeMode {
+  #[serde(rename = "message")]
+  Message,
+  #[serde(rename = "byte")]
+  Byte,
+}
+
+impl From<PipeMode> for named_pipe::PipeMode {
+  fn from(value: PipeMode) -> Self {
+    match value {
+      PipeMode::Message => named_pipe::PipeMode::Message,
+      PipeMode::Byte => named_pipe::PipeMode::Byte,
+    }
+  }
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ListenArgs {
+  path: String,
+  max_instances: Option<usize>,
+  pipe_mode: PipeMode,
+  #[serde(default = "is_true")]
+  inbound: bool,
+  #[serde(default = "is_true")]
+  outbound: bool,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ConnectArgs {
+  path: String,
+  #[serde(default = "is_true")]
+  read: bool,
+  #[serde(default = "is_true")]
+  write: bool,
+}
+
+#[op2(stack_trace)]
+#[serde]
+pub fn op_pipe_open(
+  state: &mut OpState,
+  #[serde] args: ListenArgs,
```

Comment:
A bit nitpicky, but I'd suggest to "unfurl" these argument and just specify all fields directly and not relay on serde and GC